spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-local/src/main/java/org/springframework/cloud/data/module/deployer/local/LocalModuleDeployer.java

```diff
@@ -77,9 +79,11 @@ public ModuleDeploymentId deploy(ModuleDeploymentRequest request) {
 		catch (Exception e) {
 			throw new IllegalStateException("failed to determine URL for module: " + module, e);
 		}
-		args.add(String.format("--%s.server.port=%d", module, port));
-		args.add(String.format("--%s.endpoints.shutdown.enabled=true", module));
-		launcher.launch(new String[] { module }, args.toArray(new String[args.size()]));
+		args.put("server.port", String.valueOf(port));
+		args.put("endpoints.shutdown.enabled", "true");
+
+		ModuleLaunchRequest moduleLaunchRequest = new ModuleLaunchRequest(module);
```

Comment:
Yes. The problem with `findAvailableTcpPort` is that it will always have a race condition (between the time of allocation and the time of use). That can never be completely avoided given that we are competing with other processes as well. One thing we can do to avoid a race-condition with ourselves though is to call `findAvailableTcpPort*s*(numModules)`.